tests/specifications/googlefonts_test.py

```diff
@@ -2346,19 +2343,19 @@ def test_check_162():
 
   print (f"Test FAIL with a non-RIBBI with bad nameid={NameID.TYPOGRAPHIC_SUBFAMILY_NAME} entries...")
   status, message = list(check(ttFont,
-                               style_with_spaces(font)))[-1]
+                               style(ttFont)))[-1]
   assert status == FAIL and message.code == "non-ribbi-bad-value"
 
   # remove all TYPOGRAPHIC_SUBFAMILY_NAME entries
   # by changing their nameid to something else:
   for i, name in enumerate(ttFont['name'].names):
     if name.nameID == NameID.TYPOGRAPHIC_SUBFAMILY_NAME:
       ttFont['name'].names[i].nameID = 255 # blah! :-)
-
+  
   print (f"Test FAIL with a non-RIBBI lacking a nameid={NameID.TYPOGRAPHIC_SUBFAMILY_NAME} entry...")
   status, message = list(check(ttFont,
-                               style_with_spaces(font)))[-1]
-  assert status == FAIL and message.code == "non-ribbi-lacks-entry"
```

Comment:
This is the only expected behavior change that you introduced in the code-tests.

We previously ensured the check would FAIL with a specific kind of failure and with this change you are saying that any kind of FAIL would suffice. This is relaxing the validation of the expected behavior to accept other (incorrect) log messages.

Reply:
What I mean here is that this check has got 3 different kinds of FAIL, identified by the value of `message.code`. And the code-test should be explicit on which specific FAIL a certain bad font is expected to trigger. You made the code-test less strict, which may lead to bugs not being detected in the future.

Reply:
I agree with you. Ideally I should just refactor the checks and leave the tests in place. However, if I start merging checks, I may need to do some alterations to the tests as well. I'll also run my refactored checks on the whole collection just to make sure.

Reply:
I had a video-call with Lasse (@graphicore) earlier this week in which we sketched an improved scheme for code-testing that will not require one to manually tweak the instantiation of conditions when implementing or modifying code-tests, like happened in this case. This will reduce the chances of introducing bugs on the tests themselves, by making them simpler, more concise and clearer to read / easier to write and maintain.

Well... we hope so :-D LOL

Reply:
I will open an issue about that to describe the idea in more details.